### PR TITLE
PPLUS-1478: Updated spryker-sdk/spryk

### PR DESCRIPTION
**Pull request by Kyrylo-Meretskyi**
- Developer(s): @Kyrylo-Meretskyi 
- Ticket: [PPLUS-1478](https://spryker.atlassian.net/browse/PPLUS-1478)
- Release Group: https://release.spryker.com/release-groups/view/4384
- merge: squash
#### Release Table
   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   SprykGui              | minor (currently 0.x)  |                       |
-----------------------------------------
##### Change log
Breaking Changes
- Increased `spryker-sdk/spryk` version dependency.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -5,7 +5,8 @@
     "license": "proprietary",
     "require": {
         "php": ">=7.4",
-        "spryker-sdk/spryk": "^0.3.1",
+        "spryker-sdk/spryk": "^0.4.0",
+        "spryker-sdk/spryk-src": "dev-master",
```
Comment (dereuromark) [inline composer.json:9]: U dont need this one imo

Comment (Kyrylo-Meretskyi) [reply composer.json:9]: Spryk-gui uses some classes from spryk-src
for example https://github.com/spryker-sdk/spryk-gui/blob/master/src/SprykerSdk/Zed/SprykGui/SprykGuiDependencyProvider.php#L13

Comment (dereuromark) [reply composer.json:9]: It shouldnt, as that one is becoming the phar file of spryker-sdk/spryk
cc @stereomon 

also, dev master would never be allowed here anyways.
So we need a different way of using those.

**Review by asmarovydlo — CHANGES_REQUESTED** [review]
1. Please update changelogs according to https://spryker.atlassian.net/wiki/spaces/CORE/pages/523730976/Release+Notes+and+Changelogs
2. Please fix github actions validation, it fails, but in the master it's green.

**Review by asmarovydlo — APPROVED** [review]

File: src/SprykerSdk/Zed/SprykGui/Business/Finder/Factory/FactoryInfoFinder.php [hunk 2]
```diff
@@ -94,7 +94,7 @@ protected function canReflect(string $className): bool
     {
         try {
             $betterReflection = new BetterReflection();
-            $betterReflection->classReflector()->reflect($className);
+            $betterReflection->reflector()->reflectClass($className);
```
Comment (vol4onok) [inline src/SprykerSdk/Zed/SprykGui/Business/Finder/Factory/FactoryInfoFinder.php:97]: you use roave/better-reflection package and I don't see it in composer.json

Comment (Kyrylo-Meretskyi) [reply src/SprykerSdk/Zed/SprykGui/Business/Finder/Factory/FactoryInfoFinder.php:97]: It is provided by PHPStan

**Comment by dereuromark** [comment]
Are we closing active dev on Gui @stereomon  ?
Will we deprecate the module?

**Comment by dereuromark** [comment]
Agreed
Which team is doing it?

**Comment by a-sabaa** [comment]
@stereomon why do we deprecate the GUI module? What is the replacement?

**Comment by dereuromark** [comment]
a) Broken already in prod
customers are already noticing and are annoyed

b) no maintenance, not maintainable

Normal CLI usage is supposed to be used only

**Pull request closed without merging**